kernel/src/device/fb.rs

```diff
@@ -245,6 +245,30 @@ impl FileIo for Fb {
 }
 
 impl FbHandle {
+    /// Reads an array of `u16` color map values from userspace.
+    pub fn read_color_maps_from_user(addr: usize, data: &mut [u16]) -> Result<()> {
+        if addr == 0 {
+            return Ok(());
+        }
+        for (i, item) in data.iter_mut().enumerate() {
+            let user_addr = addr + i * size_of::<u16>();
+            *item = current_userspace!().read_val(user_addr)?;
+        }
+        Ok(())
+    }
+
+    /// Writes an array of `u16` color map values to userspace.
+    pub fn write_color_maps_to_user(addr: usize, data: &[u16]) -> Result<()> {
+        if addr == 0 {
```

Comment:
In fact, we should only check this for the transparency channel. Read the Linux implementation first before making a decision: https://elixir.bootlin.com/linux/v6.17.7/source/drivers/video/fbdev/core/fbcmap.c#L281-L284

The same applies for `read_color_maps_from_user`.